### PR TITLE
Added functional tests handler and insert query test

### PR DESCRIPTION
**Pull request by Nick-S-2018**

### CODE REVIEW

**Comment by donhardman** [comment]
We should also pass tests to get it merged.

File: test/Buddy/functional/InsertQueryTest.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+use PHPUnit\Framework\TestCase;
+
+class InsertQueryTest extends TestCase {
+
+	use TestFunctionalTrait;
+
+	public function testSqlInsertQueryOk(): void {
+		echo "\nTesting the execution of SQL insert query to a non-existing table\n";
+		$query = 'INSERT into test(col1,col2) VALUES(1,2) ';
+		exec("mysql -P8306 -h127.0.0.1 -e '$query' 2>&1", $out, $res);
```
Comment (donhardman) [inline test/Buddy/functional/InsertQueryTest.php:22]: Do you think it is a good idea to test like this? What if we have no MySQL client?
Maybe it's better to test by standard endpoints of Manticore? Anyway, it works the same way if I'm right.

File: test/Buddy/functional/config/manticore.conf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+common {
```
Comment (donhardman) [inline test/Buddy/functional/config/manticore.conf:1]: This file is the same that we used in branch that I made with kill tests.

We should think about using a single file for the manticore test. So maybe we should 

I have merged that branch into main, so I guess we can arrange single place for configs in root test dir (that would be better, when configs are in same place)

File: test/src/Trait/TestFunctionalTrait.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+			}
+		}
+	}
+
+	/**
+	 * Helper that allows us to reload fresh pid for relaunched
+	 * Buddy process by the Manticore
+	 * @return void
+	 */
+	protected static function loadBuddyPid(): void {
```
Comment (donhardman) [inline test/src/Trait/TestFunctionalTrait.php:87]: This function duplicates the same one that was used in kill tests.

I have merged that branch; let's rebase and eliminate duplications of the same code.

**Review by donhardman — CHANGES_REQUESTED** [review]

File: test/Buddy/functional/InsertQueryTest.php (hunk 1 above)
Comment (Nick-S-2018) [reply test/Buddy/functional/InsertQueryTest.php:22]: I feel we need to test user's work via MySQL console too if we target the test coverage as full as possible.  But yes, it's going to require adding MySQL to the testing Docker image; it's absent now as I understand.  I think we can also add the check for MySQL availability to tests to handle such situation.

File: test/Buddy/functional/config/manticore.conf (hunk 2 above)
Comment (Nick-S-2018) [reply test/Buddy/functional/config/manticore.conf:1]: Ok, I'll do rebase from main.

File: test/Buddy/functional/InsertQueryTest.php (hunk 1 above)
Comment (donhardman) [reply test/Buddy/functional/InsertQueryTest.php]: Okay, I guess it's good point to test with different tools to be sure that everything works fine

**Review by donhardman — APPROVED** [review]
OK, lets merge

**Pull request merged**